samples/dependadotnet/Program.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using static System.Console;
+
+if (args is { Length: 0 } || args[0] is not string path)
+{
+    WriteLine("Must specify a repo root directory as input");
+    return;
+}
+
+// yaml top-matter
+
+string topMatter =
+@"# generated by dependadotnet
+# https://github.com/dotnet/core/tree/master/samples/dependadotnet
+version: 2
+updates:";
+
+WriteLine(topMatter);
+
+/* Generate the following pattern for each project file:
+
+  Note: Wednesday was chosen for quick response to .NET patch Tuesday updates
+
+- package-ecosystem: ""nuget""
+  directory: ""/"" #projectfilename
+  schedule:
+      interval: ""weekly""
+      day: ""wednesday""
+  open-pull-requests-limit: 5
+*/
+
+string regex = @"PackageReference.*Version=""[0-9]";
+
+foreach (string file in Directory.EnumerateFiles(path,"*.*",SearchOption.AllDirectories))
+{
+    if (!IsProject(file))
+    {
+        continue;
+    }
+
+
+    string filename = Path.GetFileName(file);
+    string? parentDir = Path.GetDirectoryName(file);
+    string relativeDir = parentDir?.Substring(path.Length).Replace('\\','/') ?? "/";
+    
+    bool match = false;
+    foreach (string content in File.ReadLines(file))
+    {
+        if (Regex.IsMatch(content, regex))
+        {
+            match = true;
+            break;
+        }
+    }
+
+    if (!match)
+    {
+        continue;
+    }
+
+    WriteLine( 
+$@"  - package-ecosystem: ""nuget""
+    directory: ""{relativeDir}"" #{filename}
+    schedule:
+      interval: ""weekly""
+      day: ""wednesday""
+    open-pull-requests-limit: 5");
+}
+
+bool IsProject(string filename)
+{
+    if (filename.Length < 8)
+    {
+        return false;
+    }
+
+    return filename.Substring(filename.Length - 7) switch
+    {
+        ".csproj" or ".fsproj" or ".vbproj" => true,
+        _ => false
+    };
```

Comment:
I think this can all be replaced with `Path.GetExtension`:

```suggestion
    return Path.GetExtension(filename) switch
    {
        ".csproj" or ".fsproj" or ".vbproj" => true,
        _ => false
    };
```